src/drawing/backend_impl/svg.rs

```diff
@@ -247,24 +248,36 @@ impl<'a> DrawingBackend for SVGBackend<'a> {
         if color.alpha() == 0.0 {
             return Ok(());
         }
-        let context = svg::node::Text::new(text);
-        let layout = font.layout_box(text).map_err(DrawingErrorKind::FontError)?;
 
-        let trans = font.get_transform();
-        let offset = trans.offset(layout);
-        let x0 = pos.0 + offset.0;
-        let y0 = pos.1 + offset.1;
-
-        let max_x = (layout.1).0;
-        let (dx, anchor) = match style.alignment {
-            TextAlignment::Left => (0, "start"),
-            TextAlignment::Right => (max_x, "end"),
-            TextAlignment::Center => (max_x / 2, "middle"),
+        let (x0, y0) = pos;
+        let text_anchor = match style.pos.h_pos {
+            HPos::Left => "start",
+            HPos::Right => "end",
+            HPos::Center => "middle",
         };
+
+        let ((_, x_min_y), (_, x_max_y)) =
+            font.layout_box("x").map_err(DrawingErrorKind::FontError)?;
+        let ((_, text_min_y), (_, text_max_y)) =
+            font.layout_box(text).map_err(DrawingErrorKind::FontError)?;
+        let x_height = x_max_y as f32 - x_min_y as f32;
+        let x_middle = x_min_y as f32 + x_height / 2.0;
+        let text_height = text_max_y as f32 - text_min_y as f32;
+        let text_middle = text_min_y as f32 + text_height / 2.0;
+        let scaled_height = text_height / x_height;
+        let offset = (text_middle - x_middle) / x_height;
```

Comment:
I think the offset computing has come problem. My understanding is this offset actually makes the real bottom of the bouncing box at the `pos` location, which is exactly other parts of the code defines the locations. 
If this is the case, shouldn't it be 
```rust
offset = (text_max_y - x_max_y) / x_height;
```
As long as "x" doesn't have any part under the baseline, otherwise we may ending up with half of the size between baseline and descender. 

Correct me if I was wrong.

Reply:
Also, I am ok with adding an interface that returns the relative descender size if you think this makes things easier. Since I believe rusttype have some way to returns the metrics of the font.

Reply:
You're right that the offset calculations are slightly incorrect, but it's not as simple as just using `max_y` as it makes the vertical center incorrect with the current code. I'll update the calculations and tests with the different ascends/descends. And I doubt that any interface changes are required, `layout_box` is working good enough, it's just a matter of proper testing to finalize the calculations.

Reply:
Yes, dig into this a little bit, I believe "dy=1em" is same as "dominant-baseline=hanging"
"dy=0.5ex" is same as "dominat-baseline=middle"  and
"dy=-0.5ex" is almost same as putting the dominant baseline to the actual bottom of the text.

How do you think about this. 

Also, I would like to say rusttype underestimates the text size a lot, so I think we should adjust the SVG's font size.

Reply:
To explain the magic numbera 0.75, the difference between rusttype scaling and SVG scaling is SVG uses em in pixels as font size, but rusttype uses the total height of text as the size. 
So in SVG sizing model, descender isn't a part of font size, but in rusttype it also counted as the font size. 

That gives us 1em = 1 pixel is same as rusttype size 1.3. (Assuming we have descender height of 30%. And I believe the typical descender height is some value around 20% to 30%). 
That give us the factor 1 / 1.3 = 0.76 which is almost the magic number I come up with.

Reply:
If this is the case, we actually have a pretty reliable way to define the anchor point. But the problem we have to facing is the font positioning, and my guess is this is impossible with solely SVG. 

For example, for the title spacing, we have to know the font size to decide the size of drawing area. The text size is something we have to make an reasonable guess to make it work. 

So our remaining problem could be rusttype isn't follow the SVG sizing model and that's the root for the remaining offset in the font layout. 

I actually tried render text using font-kit instead of rusttype (which I can remove a few dependency as well), it seems font-kit renders text in exact same way a browser does. Thus once we have make the change to replace rusttype, I believe we could have pretty stable SVG text layout then. 

The only thing of this plan is, changing the sizing model is definitely a breaking change which we have to wait until the next major release. 

But I believe other part of the PR seems good to me, except the label text on the right label area. 
(And I don't think this should be considered as breaking change and I believe it's ready to merge).

Sorry about the messy font handling code, as you may realized, I am just learning typography while I was creating this crate and it's a really good lesson I have learned.  

Please let me know you have any thought, and thank you so much for clean this up!

Reply:
And update, according to wikipedia, typically, the descender height is 46% of em,which give us `dy=0.76em`
This gives us pretty much accurate result, I believe in most of the case
https://jsfiddle.net/rjmtz602/

Reply:
Thank you for the research and numbers, I've updated the code and indeed it looks very good.